components/shared/net/lib.rs

```diff
@@ -933,6 +933,18 @@ pub enum NetworkError {
     SslValidation(String, Vec<u8>),
     /// Crash error, to be converted to Resource::Crash in the HTML parser.
     Crash(String),
+    UnsupportedScheme,
+    CorsViolation,
```

Comment:
I'm worried that CorsViolation covers too many different failure modes and will be hard to diagnose. Can we make it more granular?